snuba/snuba_migrations/search_issues/0001_search_issues.py

```diff
@@ -0,0 +1,97 @@
+from typing import List, Sequence
+
+from snuba.clickhouse.columns import (
+    UUID,
+    Array,
+    Column,
+    DateTime,
+    IPv4,
+    IPv6,
+    Nested,
+    String,
+    UInt,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.operations import OperationTarget, SqlOperation
+
+columns: List[Column[Modifiers]] = [
+    Column("organization_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("group_id", UInt(64)),
+    Column("search_title", String()),
+    Column("fingerprint", Array(String())),
+    Column("occurrence_id", UUID()),
+    Column("occurrence_type_id", UInt(8)),
+    Column("detection_timestamp", DateTime()),
+    Column("event_id", UUID(Modifiers(nullable=True))),
+    Column("trace_id", UUID(Modifiers(nullable=True))),
+    Column("platform", String(Modifiers(low_cardinality=True))),
+    Column("environment", String(Modifiers(low_cardinality=True, nullable=True))),
+    Column("release", String(Modifiers(low_cardinality=True, nullable=True))),
+    Column("dist", String(Modifiers(low_cardinality=True, nullable=True))),
+    Column("receive_timestamp", DateTime()),
+    Column("client_timestamp", DateTime()),
+    Column("tags", Nested([("key", String()), ("value", String())])),
+    Column("user", String(Modifiers(nullable=True))),
+    Column(
+        "user_hash", UInt(64, Modifiers(nullable=True, materialized="cityHash64(user)"))
+    ),
+    Column("user_id", UInt(64, Modifiers(nullable=True))),
+    Column("user_name", String(Modifiers(nullable=True))),
+    Column("user_email", String(Modifiers(nullable=True))),
+    Column("ip_address_v4", IPv4(Modifiers(nullable=True))),
+    Column("ip_address_v6", IPv6(Modifiers(nullable=True))),
+    Column("sdk_name", String(Modifiers(low_cardinality=True, nullable=True))),
+    Column("sdk_version", String(Modifiers(low_cardinality=True, nullable=True))),
+    Column("contexts", Nested([("key", String()), ("value", String())])),
+    Column("http_method", String(Modifiers(nullable=True, low_cardinality=True))),
+    Column("http_referer", String(Modifiers(nullable=True))),
+    Column("retention_days", UInt(16)),
+]
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    def forwards_ops(self) -> Sequence[SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=StorageSetKey.SEARCH_ISSUES,
+                table_name="search_issues_local",
+                columns=columns,
+                engine=table_engines.ReplacingMergeTree(
+                    order_by="(project_id, toStartOfDay(detection_timestamp))",
+                    partition_by="(retention_days, toMonday(detection_timestamp))",
+                    sample_by="project_id",
```

Comment:
I don't think this is a good sampling key. 
This means that, if you pass a `SAMPLE 0.5` to scan only 50% of the data the DB will scan 100% of data for 50% of the projects. Since all queries have a project id condition, it means that either sampling will be ignored or your results will be empty. 

https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree/#sample-by

In a table like this, when you sample, you tend to want a random sample of your data, so you likely want to sample by occurrence_id. or event_id if you always want all occurrences for a specific event id to either be in the sample or not.
To summarize, the sample key is generally supposed to be a high cardinality column (with a lot of different values) to get a statistically relevant sample.

Reply:
```If a sampling expression is used, the primary key must contain it. The result of a sampling expression must be an unsigned integer. Example: SAMPLE BY intHash32(UserID) ORDER BY (CounterID, EventDate, intHash32(UserID)).```

If we sample by `occurrence_id` it looks like we need to add it to the primary key. Are there concerns about adding `occurrence_id` to the last part of the primary key when most of our queries won't filter on `occurrence_id`?